atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1854,7 +1854,10 @@ bool WebContents::IsOffScreen() const {
 
 #if BUILDFLAG(ENABLE_OSR)
 void WebContents::OnPaint(const gfx::Rect& dirty_rect, const SkBitmap& bitmap) {
-  Emit("paint", dirty_rect, gfx::Image::CreateFrom1xBitmap(bitmap));
+  if (!overlay_.SendFrame(bitmap.width(), bitmap.height(), bitmap.getPixels(),
+                          bitmap.rowBytes() * bitmap.height())) {
+    Emit("paint", dirty_rect, gfx::Image::CreateFrom1xBitmap(bitmap));
```

Comment:
Shall we still emit paint event even if we directly the frame to native module?

Reply:
avoiding the js sync point is where like 80% of our performance win comes from, so i'd rather not unless there is a strong motivation to do so.